local-deploy.sh

```diff
@@ -0,0 +1,223 @@
+    echo PASS: found javac executable in PATH
+elif [[ -n "$JAVAC_HOME" ]] && [[ -x "$JAVAC_HOME/bin/javac" ]];  then
+    echo PASS: found javac executable in JAVAC_HOME     
+else
+    echo FAIL: "javac was not found on your machine"
+fi
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating maven..........................................-"
+
+if type -p mvn; then
+    echo PASS: found maven executable in PATH
+    _mvn=mvn
+elif [[ -n "$MAVEN_HOME" ]] && [[ -x "$MAVEN_HOME/bin/mvn" ]];  then
+    echo found mvn executable in MAVEN_HOME     
+else
+    echo "FAIL: maven was not found on your system"
+fi
+
+if [[ "$_mvn" ]]; then
+    version=$("$_mvn" -version 1>&1 | awk -F '"' '/Apache/ {print $0}')
+    vCompare_mvn=$(echo "${version}" 2>&1 | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}')
+    echo version: "$version"
+    if (( $(echo "$vCompare_mvn" "$minMaven" | awk '{print ($1 > $2)}') )); then
+        echo PASS: Apache Maven version is more than 3.0
+    else         
+        echo WARN: Apache Maven version is less than 3.0. Please Update.
+    fi
+fi
+
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating nodejs..........................................-"
+
+if type -p nodejs; then
+    echo PASS: found nodejs executable in PATH
+    _nodejs=nodejs
+else
+    echo "FAIL: nodejs was not found on your machine"
+fi
+
+if [[ "$_nodejs" ]]; then
+    version=$("$_nodejs" -v 2>&1 | tr -d 'v')
+    vCompare_node=$(echo "${version}" 2>&1 | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}')
+    echo version: "$version"
+    if (( $(echo "$vCompare_node" "$minNode" | awk '{print ($1 > $2)}') )); then
+        echo PASS: nodejs version is more than 4.0
+    else         
+        echo WARN: nodejs version is less than 4.0
+    fi
+fi
+if type -p npm; then
+    echo PASS: found npm executable in PATH
+    _npm=npm
+else
+    echo "FAIL: npm was not found on your machine"
+fi
+
+if [[ "$_npm" ]]; then
+    version=$("$_npm" -v )
+    echo version: "$version"
+    vCompare_npm=$(echo "${version}" 2>&1 | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}')
+    if (( $(echo "$vCompare_npm" "$minNpm" | awk '{print ($1 > $2)}') )); then
+        echo PASS: npm version is more than 3.0
+    else         
+        echo WARN: npm version is less than 3.0
+    fi
+fi
+#----------------------------------------------------------------------------------
+
+
+echo
+echo "Validating scala..........................................-"
+
+if type -p scala; then
+    echo PASS: found scala executable in PATH
+    _scala=scala
+else
+    echo "FAIL: scala was not found on your machine."
+fi
+
+if [[ "$_scala" ]]; then
+    version=$("$_scala" -version 2>&1 | awk -F ' ' '{print $5}')
+    echo version: "$version"
+    vCompare_scala=$(echo "${version}" 2>&1 | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}')
```

Comment:
FYI: for simple string deletions like this, you can use tr which is more explicit since its capabilities are more restricted:

```sh
tr -d 'A-Za-z ._-'
```